### PR TITLE
joss-paper: first stab

### PR DESCRIPTION
**Pull request by sbinet**
Updates gonum/gonum#195.

### CODE REVIEW

**Comment by sbinet** [comment]
it's just to get the ball rolling...

**Comment by kortschak** [comment]
This is too early. Note that I've raised the prospect of submitting to JOSS a couple of times on gonum-dev, but always in th context of after achieving what we consider API stability. We are not there yet (particularly after the migration to the single repo and the opportunities that gave us to break things without causing too much distress to users).

Waiting for some semblance of API stability gives us two things, a better product to go in the paper and possibly more importanly, a driver to fix up some of the API design decisions that have been noted and not had work done on them (the graph API needs work and has done for a couple of years, but I only have so much time - with @mewmew now interested we may get more work done in this direction, but probably not before the end of this semester while I am teaching).

I think there should be more discussion on gonum-dev about process before we start putting down files.

**Comment by mewmew** [comment]
> the graph API needs work and has done for a couple of years, but I only have so much time - with @mewmew now interested we may get more work done in this direction, but probably not before the end of this semester while I am teaching

I feel very enthusiastic to work on the graph API and feel we can arrive at a simple, clean, and powerful API by continued collaboration and discussions. Attending a course in graph theory at Uni this autumn, starting in a couple of weeks :)

I also feel that the merge to a single repo is a wonderful opportunity to take a careful look at the API and design of the gonum packages, before declaring them as stable.

At that stage it would be lovely to see gonum submitted to JOSS.

**Comment by sbinet** [comment]
Well, it all depends on what we expect by submitting to JOSS.
I was under the assumption that it would be a great way to raise Gonum's awareness and/or increase the number of possible developers+users.

some packages are more stable than others: perhaps we should consider submitting these instead? one paper per stable package?

anyways, I won't press this issue further.
feel free to close it.

**Comment by kortschak** [comment]
Sure, really all I would like is that there be some discussion first, preferably on gonum-dev rather than here.

**Comment by kortschak** [comment]
Also, it may be worth reaching out to the editorial board of JOSS to see what kinds of models are appropriate. It is unclear to me whether it is possible to submit per package multiple times for a single repo, and gonom/gonum now is substantially bigger than the common size of works submitted from a recent survey I took.

File: joss-paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Summary
+
+Gonum provides tools enabling scientific research through consistent,
+composable and comprehensible scientific code.
+Gonum aims to be the equivalent of NumPy/SciPy but for the Go programming
+language.
+
+Gonum strives to deliver code that run fast (as fast as C/C++ or Fortran)
+and code that is as easy to write than in Matlab, R or Python.
```
Comment (mewmew) [inline joss-paper/paper.md]:
```diff
-as easy to write than in Matlab, R or Python
+as easy to write as in Matlab, R or Python
```

Comment (sbinet) [inline joss-paper/paper.md]: done

**Comment by btracey** [comment]
I agree with Dan that jumping to PRs already is premature. There is a lot to be discussed (location, scope, etc.). My past experience with writing these kind of software papers is that it's not necessary to have perfectly stable APIs, but having a stable location (which we now have) is important. I'm unfortunately gone for a week starting now, but I'll be happy to continue the discussion then.

File: joss-paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+title: 'Gonum: consistent, composable and comprehensible scientifc code'
```
Comment (Konstantin8105) [inline joss-paper/paper.md]: misspell in word "scientifc"

**Comment by kortschak** [comment]
I think we might soon be in a position to start work on this again. Where do we want discussion to happen?

**Comment by soypat** [comment]
Was this ever done? I've found sbinet's Go-HEP paper but nothing else.

**Comment by sbinet** [comment]
nope.
IIRC, the consensus was to wait for a `v1.0.0`.
(let me close this one.)

**Pull request closed without merging**